apio-consumer/src/main/kotlin/com/liferay/apio/consumer/parser/ThingParser.kt

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.apio.consumer.parser
+
+import com.google.gson.Gson
+import com.google.gson.JsonParseException
+import com.google.gson.JsonSyntaxException
+import com.google.gson.reflect.TypeToken
+import com.liferay.apio.consumer.exception.ApioException
+import com.liferay.apio.consumer.exception.CantParseToThingException
+import com.liferay.apio.consumer.graph.ApioGraph
+import com.liferay.apio.consumer.model.*
+import okhttp3.Response
+
+/**
+ * @author Javier Gamarra
+ */
+class ThingParser {
+
+    companion object {
+        @JvmStatic
+        @Throws(CantParseToThingException::class)
+        fun parse(response: Response): Thing {
+            return response.body()?.let {
+                parse(it.string())
+            }?.let {
+                val (thing, embeddedThings) = it
+                ApioGraph.updateNodes(thing, embeddedThings)
+
+                if (!response.isSuccessful) {
```

Comment:
This seems worth checking after tryingt to parse it, right?

Reply:
oops my bad, I'll fix that.

Reply:
In this case we would have to parse the response exception. I'll fix in the next commits.